docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+  * `autorenew.gracePeriod`
+- Each Hedera entity has an `expirationTime` which is the effective consensus timestamp at (and after) which the entity is set to expire.
+- Each Hedera entity also has an `autoRenewAccount` which is the account to pay for the fee at renewal. This `autoRenewAccount` could be itself if the entity is a crypto account or an account associated with the entity when it was created.
+- When a Hedera entity is created, its initial lifetime is defined by its `autoRenewPeriod`. At its `expirationTime`, Hedera Services will try to extend an entity's lifetime by another `autoRenewPeriod` if the `autoRenewAccount` has enough balance to do so, or as much extension as the remaining balance permits.
+- After handling a transaction, Hedera Services will search within the next `autorenew.numberOfEntitiesToCheck` for upto `autorenew.maxNumberOfEntitiesToRenew` entities that expired then try to renew these entities.
+- After the grace period, if the `expirationTime` of an entity is not extended, it will be deleted from the system.
+
+## Implementation
+Hedera Services will perform a circular scanning of entities, meaning after we reach the last entity in the system, we will go back scanning from the first entity in the system.
+
+When trying to renew an entity:
+1. Calculate the fee to extend the entity's `expirationTime` for another `autoRenewPeriod`.
+2. If the `autoRenewAccount` of the entity has enough balance to cover this fee:
+  - extend the entity's `expirationTime` for another `autoRenewPeriod`.
+  - otherwise, translate the remaining balance of the `autoRenewAccount` into an extension, preferably proportional to the fee calculated in step 1, then extend accordingly.
+3. If the grace period also passes, delete the entity from the system.
+
+For restart and reconnect: The last scanned entity must be in the state for synchronization.
+
+## Autorenewal record
+After autorenewing an entity, Hedera Services will generate a `TransactionRecord` that serves as an autorenewal record and contains the following:
+
+| Field | Type | Label | Description |
+|---|---|---|---|
+| receipt | TransactionReceipt | | receipt will contain either an accountID, a fileID, a contractID, a topicID or a tokenID that got autorenewed |
+| transactionHash | bytes | | empty |
+| consensusTimestamp | Timestamp | | The consensus timestamp of the autorenewal |
+| transactionID | TransactionID | | { empty transactionValidStart, autoRenewAccount } |
+| memo | string | | "Entity {ID} was automatically renewed. New expiry: {newExpiry}" |
+| transactionFee | uint64 | | The fee charged for the autorenewal of the entity |
+| contractCallResult | ContractFunctionResult | | empty |
+| contractCreateResult | ContractFunctionResult | | empty |
+| transferList | TransferList | | {(autoRenewAccount, -transactionFee), (defaultFeeCollectionAccount, transactionFee)} |
+| tokenTransferLists | TokenTransferList | repeated | empty |
+| scheduleRef | ScheduleID | | empty |
```

Comment:
`body` is missing. Is the intent to mark it as empty?
Is there a possibility to have the body be a crypto transfer.
This would be important for an accounts balance transaction history to see how the balance changes over time - just like you might see bank service fees in your bank account.